compiler/liftdestructors.nim

```diff
@@ -851,6 +869,8 @@ proc ownedClosureOp(c: var TLiftCtx; t: PType; body, x, y: PNode) =
   of attachedDeepCopy: assert(false, "cannot happen")
   of attachedTrace: discard
   of attachedWasMoved: body.add genBuiltin(c, mWasMoved, "wasMoved", x)
+  of attachedDup:
+    assert false, "cannot happen"
```

Comment:
In follow-up PRs we should lift `=dup` like other hooks.

Reply:
@araq Hi, may I ask what's the benefit of lifting an operation? Like it can be used by other hooks or something?

Reply:
The `=dup` hook is more efficient than the `wasMoved(tmp); =sink(tmp, y)` combination in principle. If you lift it so that more types are affected it remains to be more efficient.